demo/lib/vm.bash

```diff
@@ -174,6 +174,49 @@ vm-check-running-binary() {
     return 0
 }
 
+vm-wait-process-startup() {
+    # Usage: vm-wait-process-startup [--timeout TIMEOUT] [--pidfile PIDFILE] PROCESS
+    # Wait for a PROCESS (string) to appear in process list (ps -A output).
+    # If pidfile parameter is given, we also check that the process has the file open.
+    # The default TIMEOUT is 30 seconds.
+
+    local process timeout pidfile invalid
+
+    timeout=30
+    while [ "${1#-}" != "$1" ] && [ -n "$1" ]; do
+        case "$1" in
+            --timeout)
+                timeout="$2"
+                shift; shift
+                ;;
+            --pidfile)
+                pidfile="$2"
+                shift; shift
+                ;;
+            *)
+                invalid="${invalid}${invalid:+,}\"$1\""
+                shift
+                ;;
+        esac
+    done
+    if [ -n "$invalid" ]; then
+        error "invalid options: $invalid"
+        return 1
+    fi
+    process="$1"
+
+    # As we first wait for the process, and then wait for the pidfile (if enabled)
+    # we might wait longer than expected. Accept that anomaly atm.
+    vm-wait-process --timeout $timeout "$process" || command-error "timeout while waiting $process"
```

Comment:
Use `error` rather than `command-error` in all occurrences in this function.

`command-error` works only for reporting an error from previous command execution that is visible to the user, that is, command execution that is wrapped with `command-start` and `command-end`. As a rule of thumb, it works immediately after `vm-command`, but elsewhere almost certainly `error` is a better choice.